Userland/Libraries/LibWeb/HTML/WindowOrWorkerGlobalScope.cpp

```diff
@@ -554,4 +553,18 @@ void WindowOrWorkerGlobalScopeMixin::queue_the_performance_observer_task()
     });
 }
 
+// https://html.spec.whatwg.org/multipage/timers-and-user-prompts.html#run-steps-after-a-timeout
+void WindowOrWorkerGlobalScopeMixin::run_steps_after_a_timeout(i32 timeout, Function<void()> completion_step)
+{
+    auto id = m_timer_id_allocator.allocate();
+    return run_steps_after_a_timeout_impl(timeout, move(completion_step), id);
+}
+
+void WindowOrWorkerGlobalScopeMixin::run_steps_after_a_timeout_impl(i32 timeout, Function<void()> completion_step, i32 id)
+{
+    auto timer = Timer::create(this_impl(), timeout, move(completion_step), id);
+    m_timers.set(id, timer);
+    timer->start();
+}
```

Comment:
Given that the spec text is:
```c++
and an optional value timerKey:
```

I think we need to implement this as one function, and make the timer key an Optional:
```c++
// header
void run_steps_after_a_timeout_impl(i32 timeout, Function<void()> completion_step, Optional<i32> timer_key = {});

// impl
// https://html.spec.whatwg.org/multipage/timers-and-user-prompts.html#run-steps-after-a-timeout
void WindowOrWorkerGlobalScopeMixin::run_steps_after_a_timeout_impl(i32 timeout, Function<void()> completion_step, Optional<i32> timer_key)
{
    // 2. If timerKey is not given, then set it to a new unique non-numeric value.
    if (!timer_key.has_value())
        timer_key = m_timer_id_allocator.allocate();
}
```

Reply:
I've kept this as 2 functions in order to enforce step 1:
> 1. Assert: if timerKey is given, then the caller of this algorithm is the timer initialization steps. (Other specifications must not pass timerKey.)

I've refactored it to use an Optional as suggested. It seems a bit unnecessary in this case, but it does more closely follow the spec.